tamagotchi/src/lib.rs

```diff
@@ -133,9 +135,20 @@ async fn main() {
             debug!("Successfully approved tokens");
             msg::reply(approval_result, 0).expect("Error sending approval result");
         }
-        TmOwner::Owner => {
+        TmAction::Owner => {
             msg::reply(TmEvent::Owner(tamagotchi.owner), 0).expect("reply failed on owner");
         }
+        TmAction::CheckState => {
+            tamagotchi.check_state_flow();
+        }
+        TmAction::ReserveGas {
+            reservation_amount,
+            duration,
+        } => {
+            msg::reply(tamagotchi.make_reservation(reservation_amount, duration), 0)
```

Comment:
I'd check whether `amount` is sufficient for one handling and `duration` is not too short. Just for information, there is the [`Reservation`](https://docs.gear.rs/gstd/struct.Reservation.html) struct that keeps some additional info about reservation internally.